### PR TITLE
[dot/rpc] implement RPC system_accountNextIndex

### PR DESCRIPTION
**Pull request by edwardmack**
## Changes

- Implement system_accountNextIndex to lookup account nonce from state storage.
-
-

## Tests

```
go test ./dot/rpc/... -run TestSystemModule_AccountNextIndex
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [ ] All integration tests and required coverage checks are passing

## Issues

- related to #731 and #265

### CODE REVIEW

**Review by RyRy79261 — APPROVED** [review]

**Review by arijitAD — APPROVED** [review]

File: dot/rpc/modules/system.go [hunk 1]
```diff
@@ -135,3 +151,47 @@ func (sm *SystemModule) NodeRoles(r *http.Request, req *EmptyRequest, res *[]int
...
+		return err
+	}
+	var metadata types.Metadata
+	err = types.DecodeFromBytes(sdMeta.([]byte), &metadata)
+	if err != nil {
+		return err
+	}
+
+	if req == nil || len(req.String) == 0 {
+		return errors.New("Account address must be valid")
```
Comment (noot) [inline dot/rpc/modules/system.go:175]: maybe move this to the top of the function so it checks it first?

**Review by noot — APPROVED** [review]
looks good!

File: dot/rpc/modules/system.go (hunk 1 above)
Comment (edwardmack) [reply dot/rpc/modules/system.go]: good idea.

**Comment by edwardmack** [comment]
Added check of pending transactions before looking in storage.  codeclimate is complaining, so please take a re-look at this and let me know suggestions for cleaning it up.

**Review by RyRy79261 — APPROVED** [review]

File: dot/rpc/modules/system.go [hunk 2]
```diff
@@ -135,3 +157,74 @@ func (sm *SystemModule) NodeRoles(r *http.Request, req *EmptyRequest, res *[]int
...
+		}
+		extSigner, err := common.HexToBytes(fmt.Sprintf("0x%x", ext.Signature.Signer.AsAccountID))
+		if err != nil {
+			return err
+		}
+		if bytes.Equal(extSigner, addressPubKey) {
+			found = true
+			sigNonce := big.Int(ext.Signature.Nonce)
+			if sigNonce.Uint64() > nonce {
+				nonce = sigNonce.Uint64()
```
Comment (noot) [inline dot/rpc/modules/system.go:186]: wouldn't the nonce always be greater unless it's 0, in which case it's fine to directly set it?

File: dot/rpc/modules/system_test.go [hunk 3]
```diff
@@ -230,10 +233,73 @@ func TestSystemModule_Version(t *testing.T) {
...
+
 	expected := map[string]interface{}(nil)
+
 	res := new(interface{})
 	err := sys.Properties(nil, nil, res)
 	require.NoError(t, err)
 	require.Equal(t, expected, *res)
 }
+
+func TestSystemModule_AccountNextIndex(t *testing.T) {
```
Comment (noot) [inline dot/rpc/modules/system_test.go:246]: is it possible to add a test for the non-pending case?

**Review by noot — APPROVED** [review]
nice work!

File: dot/rpc/modules/system.go (hunk 2 above)
Comment (edwardmack) [reply dot/rpc/modules/system.go:186]: Yes, but there may be more than one tx signed by that user in the tx pool, so did this to make sure we have the maximum nonce.

File: dot/rpc/modules/system_test.go (hunk 3 above)
Comment (edwardmack) [reply dot/rpc/modules/system_test.go]: Yes, I've added tests.

**Pull request merged**